proto_parser/src/proto_parser.cc

```diff
@@ -0,0 +1,114 @@
+/*
+*   Copyright 2020 Google LLC
+*
+*   Licensed under the Apache License, Version 2.0 (the "License");
+*   you may not use this file except in compliance with the License.
+*   You may obtain a copy of the License at
+*
+*    https://www.apache.org/licenses/LICENSE-2.0
+*
+*   Unless required by applicable law or agreed to in writing, software
+*   distributed under the License is distributed on an "AS IS" BASIS,
+*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*   See the License for the specific language governing permissions and
+*   limitations under the License.
+*/
+
+#include "proto_parser.h"
+#include "message_value.h"
+#include "primitive_value.h"
+
+namespace wireless_android_play_analytics {
+
+void ProtoParser::PopulateFields(int& prev_field_line, 
+    const google::protobuf::TextFormat::ParseInfoTree& tree,
+    const google::protobuf::Message& message,
+    ProtoValue* proto_value, int indent_count) {
+  //TODO(alexanderlin): Add Unit test.
+  const google::protobuf::Reflection* reflection = message.GetReflection();
+  const google::protobuf::Descriptor* descriptor = message.GetDescriptor();
+  MessageValue* message_tmp = 
```

Comment:
Why not passing in MessageValue* (instead of ProtoValue*) in the function if proto_value will be casted into MessageValue* anyway.

Reply:
it's so that all the functions are consistent in that they all take protovalue* as input and do the conversion themselves instead of having the caller do the conversion.